cmd/sidecar/main.go

```diff
@@ -33,26 +41,54 @@ func main() {
 		os.Exit(1)
 	}
 
+	if configAuthn.ContainerMode == "init" {
+		os.Exit(0)
+	}
+
 	// Create new Authenticator
 	authn, err := authenticator.New(*configAuthn)
 	if err != nil {
 		errLogger.Printf(err.Error())
 		os.Exit(1)
 	}
 
-	configSecrets, err := secretsConfig.NewFromEnv(tokenFilePath)
+	// Create new Storage
+	configStorage, err := storageConfig.NewFromEnv()
 	if err != nil {
 		errLogger.Printf(err.Error())
 		os.Exit(1)
 	}
-
-	// Create new Secrets
-	secrets, err := secrets.New(*configSecrets)
+	stor, err := storageHandler.NewStorageHandler(*configStorage)
 	if err != nil {
 		errLogger.Printf(err.Error())
 		os.Exit(1)
 	}
 
+	// TODO: account for if SECRET_DESTINATION is set to k8s but not an init container. Must check this case early.
+	conjurSecretsDest := none
+	/* if conjurSecretsDest == 1  &&  authn.Config.ContainerMode != "init" {
+	     // notify on invalid configuration and exit
+	     infoLogger.Printf("Appropriate message for not supporting sidecar with this workflow ....")
+	     os.Exit(1)
+	} */
+
+	// TODO: move these code segments to their respective files
+	//  Code that will handle the placement of access token
+	if conjurSecretsDest == none {
+		configSecrets, err := secretsConfig.NewFromEnv(tokenFilePath)
+		if err != nil {
+			errLogger.Printf("Failure creating secrets config: %s", err.Error())
+			os.Exit(1)
+		}
+
+		// Create new Secrets
+		secretsHandler, err = secrets.New(*configSecrets)
+		if err != nil {
+			errLogger.Printf("Failure creating secrets: %s", err.Error())
```

Comment:
do I have to put a this error code block whenever we expect an error to be returned? @orenbm

Reply:
as long as we think it can assist us